Lib/asyncio/timeouts.py

```diff
@@ -34,9 +35,21 @@ def __init__(self, when: Optional[float]) -> None:
         self._when = when
 
     def when(self) -> Optional[float]:
+        """Return the current deadline, or `None` if the current deadline is not set.
+
+        The deadline is a float, consistent with the time returned by `loop.time()`.
+        """
         return self._when
 
     def reschedule(self, when: Optional[float]) -> None:
+        """Change the time at which the timeout will trigger.
+
+        - If `when` is `None`, any current deadline will be removed,
+            and the context manager will wait indefinitely.
+        - If `when` is a float, it is set as the new deadline.
+        - if `when` is in the past, the timeout will trigger on the
+            next iteration of the event loop.
```

Comment:
This is generally true, not just for this API, so maybe mention that in the constructor docstring.